code/lib/cli/src/automigrate/fixes/new-frameworks/new-frameworks.test.ts

```diff
@@ -0,0 +1,715 @@
+/* eslint-disable no-underscore-dangle */
+import * as path from 'path';
+import type { StorybookConfig } from '@storybook/types';
+import * as coreCommon from '@storybook/core-common';
+import type { JsPackageManager, PackageJson } from '../../../js-package-manager';
+import * as rendererHelpers from '../../helpers/detectRenderer';
+import * as frameworkHelpers from '../../helpers/detectFramework';
+
+import { newFrameworks } from './new-frameworks';
+
+jest.mock('@storybook/core-common', () => ({
+  ...jest.requireActual('@storybook/core-common'),
+  loadMainConfig: jest.fn(),
+}));
+
+// eslint-disable-next-line global-require, jest/no-mocks-import
+jest.mock('fs-extra', () => require('../../../../../../__mocks__/fs-extra'));
+
+const checkNewFrameworks = async ({
+  packageJson,
+  main,
+}: {
+  packageJson: PackageJson;
+  main: Partial<StorybookConfig> & Record<string, unknown>;
+}) => {
+  if (main) {
+    jest.spyOn(coreCommon, 'loadMainConfig').mockReturnValue(Promise.resolve(main) as any);
+    // eslint-disable-next-line global-require
+    require('fs-extra').__setMockFiles({
+      [path.join('.storybook', 'main.js')]: `
+        const config = ${JSON.stringify(main)};
+        export default config;
+      `,
+    });
+  } else {
+    jest
+      .spyOn(coreCommon, 'loadMainConfig')
+      .mockReturnValue(Promise.reject(new Error('could not find main.js!')));
+  }
+  const packageManager = {
+    retrievePackageJson: () => ({ dependencies: {}, devDependencies: {}, ...packageJson }),
+  } as JsPackageManager;
+  return newFrameworks.check({ packageManager });
+};
+
+describe('new-frameworks fix', () => {
```

Comment:
I'm adding a comment here so this file doesn't get overlooked. Github is collapsing it by default